### ISSUE TITLE
Issues with colors or custom function color handling?

### ISSUE THREAD

**Issue by thesjg** [leafo/scssphp#436]
Using the most recent release, 0.6.3

Here is what I am trying to do, I have a scss file containing:

```
$brand-primary:             dreamerysetting('origin_theme_color_brand_primary');
@import 'bootstrap';
```

dreamerysetting is registered with the scss compiler using registerFunction, and returns a hex color string, like "#ffcd03" (sans quotes). The version of bootstrap being imported is 4alpha2, so $brand-primary is defined as:

`$brand-primary:             #0275d8 !default;`

This part seems to work fine, but later on in bootstrap's _variables.scss the following happens:

```
$link-color:                 $brand-primary !default;
$link-hover-color:           darken($link-color, 15%) !default;
```

The call to darken asserts in libDarken because the "color" of $link-color has an internal type of "keyword", not "color". I temporarily fixed this in my tree by modifying coerceColor as follows (this may not be the right approach):

```
            case Type::T_KEYWORD:
                $name = strtolower($value[1]);
                if (isset(Colors::$cssColors[$name])) {
                    $rgba = explode(',', Colors::$cssColors[$name]);

                    return isset($rgba[3])
                        ? [Type::T_COLOR, (int) $rgba[0], (int) $rgba[1], (int) $rgba[2], (int) $rgba[3]]
                        : [Type::T_COLOR, (int) $rgba[0], (int) $rgba[1], (int) $rgba[2]];
                } else if (substr($name, 0, 1) == '#') {
                    /* Doesn't accommodate short hex colors */
                    $rgba = array();
                    $rgba[] = hexdec(substr($name, 1, 2));
                    $rgba[] = hexdec(substr($name, 3, 2));
                    $rgba[] = hexdec(substr($name, 5, 2));
                    return [Type::T_COLOR, (int) $rgba[0], (int) $rgba[1], (int) $rgba[2]];
                }

                return null;
```

**Comment by robocoder** [comment]
I've committed a fix in b42b2e23128b06f482f8947533f7dfa0c49a3123

Note: if your custom function doesn't specify the return value's type, then it is coerced to a keyword.  (See Compiler::coerceValue.)

An alternate solution is for your custom function to return `[Type::T_COLOR, 0x02, 0x75, 0xd9]` instead of `'#0275d9'`. This approach has the added capability of being able to specify the opacity.

**Comment by thesjg** [comment]
Excellent -- it wasn't clear from the documentation (
http://leafo.net/scssphp/docs/#custom_functions ) the return values should
be typed, nor was it clear how to type them. A small addition to the
documentation would make this a lot more clear.

Thanks for the fix as well as the insight!
